sql/core/benchmarks/DataSourceReadBenchmark-results.txt

```diff
@@ -1,252 +1,275 @@
 ================================================================================================
 
-OpenJDK 64-Bit Server VM 1.8.0_282-b08 on Linux 5.4.0-1043-azure
-Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
+OpenJDK 64-Bit Server VM 1.8.0_312-b07 on Linux 5.11.0-1020-azure
+Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz
 SQL Single TINYINT Column Scan:           Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-SQL CSV                                           15943          15956          18          1.0        1013.6       1.0X
-SQL Json                                           9109           9158          70          1.7         579.1       1.8X
-SQL Parquet Vectorized                              168            191          16         93.8          10.7      95.1X
-SQL Parquet MR                                     1938           1950          17          8.1         123.2       8.2X
-SQL ORC Vectorized                                  191            199           6         82.2          12.2      83.3X
-SQL ORC MR                                         1523           1537          20         10.3          96.8      10.5X
-
-OpenJDK 64-Bit Server VM 1.8.0_282-b08 on Linux 5.4.0-1043-azure
-Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
+SQL CSV                                           16820          16859          54          0.9        1069.4       1.0X
+SQL Json                                          11583          11586           4          1.4         736.4       1.5X
+SQL Parquet Vectorized                              164            177          11         96.0          10.4     102.7X
+SQL Parquet MR                                     2839           2857          25          5.5         180.5       5.9X
+SQL ORC Vectorized                                  150            161           7        104.8           9.5     112.1X
+SQL ORC MR                                         1915           1923          12          8.2         121.7       8.8X
+
+OpenJDK 64-Bit Server VM 1.8.0_312-b07 on Linux 5.11.0-1020-azure
+Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz
 Parquet Reader Single TINYINT Column Scan:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 -------------------------------------------------------------------------------------------------------------------------
-ParquetReader Vectorized                             203            206           3         77.5          12.9       1.0X
-ParquetReader Vectorized -> Row                       97            100           2        161.6           6.2       2.1X
+ParquetReader Vectorized                             211            218           5         74.6          13.4       1.0X
+ParquetReader Vectorized -> Row                      286            293           7         55.1          18.2       0.7X
```

Comment:
Interesting, a similar pattern for TINYINT.

Reply:
Yea not sure what's going on, but highly doubt it's related to this PR though. Also "ParquetReader Vectorized" stays the same so it's related to related to the columnar -> row path

Reply:
Yes, you are right, it is likely unrelated but I think we might need to take a look into this, maybe it is just noise in the benchmark results but the drop is quite significant though.

Reply:
Same here.

Reply:
It looks like noise, in the new run:
```
Parquet Reader Single TINYINT Column Scan:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
-------------------------------------------------------------------------------------------------------------------------
ParquetReader Vectorized                             179            185           9         88.0          11.4       1.0X
ParquetReader Vectorized -> Row                       91            101           3        172.6           5.8       2.0X
```

but come to think about it, I wonder how this makes sense: shouldn't `ParquetReader Vectorized -> Row` always be more expensive than `ParquetReader Vectorized`, it basically does the same thing as latter but with the extra columnar to row conversion.